doc/ntp/ntp-design.md

```diff
@@ -0,0 +1,1118 @@
+| server         | server, pool           | NTP_SERVER\|\<key\>                |
+| state (server) | N/A                    | NTP_SERVER\|\<key\>\|admin_state:\<val\> |
+| type           | server, pool           | NTP_SERVER\|\<key\>\|association_type:\<val\> |
+| version        | version                | NTP_SERVER\|\<key\>\|version:\<val\> |
+| aggressive     | iburst                 | NTP_SERVER\|\<key\>\|iburst:\<val\>|
+| key (server)   | key                    | NTP_SERVER\|\<key\>\|key:\<val\>   |
+| key (auth)     | keyno                  | NTP_KEY\|\<key\>\|value:\<val\>    |
+| type (auth)    | type                   | NTP_KEY\|\<key\>\|type:\<val\>     |
+| trusted        | trustedkey             | NTP_KEY\|\<key\>\|trusted:\<val\>  |
+
+<!-- omit in toc -->
+### 2.3.3 NTP configuration
+NTP global configuration offers `state`, `authentication`, `vrf` parameters for flexible configuration of NTP client.
+For the server the next configuration options are supported: `state`, `type`, `version`, `key`.
+And finally NTP key management supports the next options: `value`, `type`, `trusted`.
+
+**Note:**
+1. The destination might be not reachable over the specified `vrf`/`host`: no way to check - user's responsibility
+2. Feature should handle automatically OOB VRF changes.
+
+<!-- omit in toc -->
+#### 2.3.3.1 Server configurations: `add/del/change`
+Here is a part of `ntp.conf.j2` template that can be used to generate configured NTP servers:
+
+**Example:**
+```jinja
+{# Getting NTP global configuration -#}
+{% set global = (NTP | d({})).get('global', {}) -%}
+
+{# Adding NTP servers. We need to know if we have some pools, to set proper
+config -#}
+{% set is_pools = False %}
+{% for server in NTP_SERVER if NTP_SERVER[server].admin_state != 'disabled' and
+                               NTP_SERVER[server].resolve_as and
+                               NTP_SERVER[server].association_type -%}
+    {% set config = NTP_SERVER[server] -%}
+    {# Server options -#}
+    {% set soptions = '' -%}
+    {# Server access control options -#}
+    {% set aoptions = '' -%}
+
+    {# Authentication key -#}
+    {% if config.key -%}
+        {% set soptions = soptions ~ ' key ' ~ config.key -%}
+    {% endif -%}
+
+    {# Aggressive polling -#}
+    {% if config.iburst -%}
+        {% set soptions = soptions ~ ' iburst' -%}
+    {% endif -%}
+
+    {# Protocol version -#}
+    {% if config.version -%}
+        {% set soptions = soptions ~ ' version ' ~ config.version -%}
+    {% endif -%}
+
+    {# Check if there are any pool configured. BTW it doesn't matter what was
+    configured as "resolve_as" for pools. If they were configured with FQDN they
+    must remain like that -#}
+    {% set config_as = config.resolve_as -%}
+    {% if config.association_type == 'pool' -%}
+        {% set is_pools = True -%}
+        {% set config_as = server -%}
+    {% else -%}
+        {% set aoptions = aoptions ~ ' nopeer' -%}
+    {% endif -%}
```

Comment:
There are tabs in this area. The file view has indentation issues. Please use white spaces only.

Reply:
No tabs here. I indented that file like that for better readability